play-json/shared/src/main/scala/play/api/libs/json/Format.scala

```diff
@@ -32,26 +32,27 @@ object OFormat {
 
   implicit def GenericOFormat[T](implicit fjs: Reads[T], tjs: OWrites[T]): Format[T] = apply(fjs, tjs)
 
-  def apply[A](read: JsValue => JsResult[A], write: A => JsObject): OFormat[A] = new OFormat[A] {
+  def apply[A](read: JsValue => JsResult[A], write: A => JsObject): OFormat[A] =
+    new FunctionalOFormat[A](read, write)
```

Comment:
Use specific impl class, not to exclude null-check for instances created by these factories